### PR TITLE
Change expiration.RenewToken's response from logical.Auth to logical.Response

### LINKED ISSUE #1164: Github auth: Token still working even after validity check fails
I was very excited to see https://github.com/hashicorp/vault/pull/1047 from @vishalnayak in 0.5.1.  However, I'm not seeing what I would have expected here.  My test:

1. I'm in a github team that allows me access to stuff.
2. I do `vault auth -github ...`, and can `vault list /secret` or whatever else I should have access to.
3. I remove myself from said Github team.  I should now have no access to vault via github teams.
4. Just as a note, yes, I understand that after removing from github but prior to attempting to renew, it will still give me access (and it does).
5. I do `vault token-renew ...` and the output is simply the `Key	Value` header.  So it does detect that something has changed.
6. At this point, I've done a token renew, so I would expect future calls to things (`vault list /secret` or whatever) should fail.  It does not.  Everything still behaves as if my token was valid.

Is there a step missing (or not working) to revoke the token if the validity check fails?

### PR DESCRIPTION
**Pull request by vishalnayak**
All tests are passing.
Fixes #1164

### CODE REVIEW

File: vault/expiration.go [hunk 1]
```diff
@@ -332,12 +332,16 @@ func (m *ExpirationManager) RenewToken(req *logical.Request, source string, toke
 		return nil, err
 	}
 
-	// Fast-path if there is no renewal
 	if resp == nil {
 		return nil, nil
 	}
+
+	if resp.IsError() {
+		return resp, nil
```
Comment (jefferai) [inline vault/expiration.go]: I think what we should do here is actually not trust what the backend gave us, but instead create a new logical.Response and copy the error from `resp` into the new one, then return that.

File: vault/expiration.go [hunk 2]
```diff
@@ -332,12 +332,16 @@ func (m *ExpirationManager) RenewToken(req *logical.Request, source string, toke
...
+	}
+
 	if resp.Auth == nil || !resp.Auth.LeaseEnabled() {
-		return resp.Auth, nil
+		return resp, nil
```
Comment (jefferai) [inline vault/expiration.go]: Same here.

File: vault/expiration.go [hunk 3]
```diff
@@ -354,7 +358,7 @@ func (m *ExpirationManager) RenewToken(req *logical.Request, source string, toke
 
 	// Update the expiration time
 	m.updatePending(le, resp.Auth.LeaseTotal())
-	return resp.Auth, nil
+	return resp, nil
```
Comment (jefferai) [inline vault/expiration.go]: Also here. :-)

File: vault/expiration.go (hunk 1 above)
Comment (vishalnayak) [inline vault/expiration.go]: Okay. I'll copy the Data part as-is.

File: vault/expiration.go (hunk 2 above)
Comment (vishalnayak) [inline vault/expiration.go]: I don't see an option but to only create an empty response object here.

File: vault/expiration.go (hunk 3 above)
Comment (vishalnayak) [inline vault/expiration.go]: I'll copy only the Auth object here since error case is previously handled.

File: vault/expiration.go [hunk 4]
```diff
@@ -332,12 +332,18 @@ func (m *ExpirationManager) RenewToken(req *logical.Request, source string, toke
...
+
+	if resp.IsError() {
+		return &logical.Response{
+			Data: resp.Data,
+		}, nil
+	}
+
 	if resp.Auth == nil || !resp.Auth.LeaseEnabled() {
-		return resp.Auth, nil
+		return &logical.Response{}, nil
```
Comment (jefferai) [inline vault/expiration.go]: In this case I'd go with the previous behavior -- set `Auth` in the new `logical.Response` to resp.Auth`. If it's `nil`, no harm no foul; if it's not, other information will continue through.

**Comment by jefferai** [comment]
Made one more comment -- other than that, LGTM!

**Pull request merged**